src/lib/booleans.js

```diff
@@ -0,0 +1,118 @@
+function negate(a) {
+  return !a;
+}
+
+function both(a, b) {
+  return a && b;
+}
+
+function either(a, b) {
+  return a || b;
+}
+
+function none(a, b) {
+  return !a && !b;
+}
+
+function one(a, b) {
+  if (!a && !b) {
+    return false;
+  }
+  return !a || !b;
+}
+function truthiness(a) {
+  if (a === 0) {
+    return false;
+  }
+  if (a === '') {
+    return false;
+  }
+  if (!a === !NaN) {
+    return false;
+  }
+  if (a == null) {
+    return false;
+  }
+  return true;
+}
+
+function isEqual(a, b) {
+  if (a === b) {
+    return true;
+  }
+  return false;
+}
+
+function isGreaterThan(a, b) {
+  if (a > b) {
+    return true;
+  }
+  return false;
+}
+
+function isLessThanOrEqualTo(a, b) {
+  if (a <= b) {
+    return true;
+  }
+  return false;
+}
+
+function isOdd(a) {
+  if (a % 2 !== 0) {
+    return true;
+  }
+  return false;
+}
+
+function isEven(a) {
+  if (a % 2 === 0) {
+    return true;
+  }
+  return false;
+}
+
+function isSquare(a) {
+  if (Math.sqrt(a) % 1 === 0) {
+    return true;
+  }
+  return false;
+}
+
+function startsWith(char, string) {
+  if (string.startsWith(char) === true) {
+    return true;
+  }
+  return false;
+}
+
+function containsVowels(string) {
+  if (string.match('a|A|e|E|i|I|o|O|u|U')) {
+    return true;
+  }
+  return false;
```

Comment:
So this feels like a good chance to introduce regex to you!
This is one of those things that you'll end up using at some point, its a very versatile and powerful way of matching patterns in strings very quickly. They are written in a format like `/[aeiou]/`, and once you get the hang of them they are really useful

What this example means is: match any part of a given string with any character between the square brackets. In this case it will only match these exact characters in lower case, but a small modifier of the `i` flag at the end (making it `/[aeiou]/i`) makes it case insensitive, so will match both cases of anything in the brackets.
These are used with the `.match` js function, like this:
```
someString.match(/[aeiou]/i)
```
this will return either true if it finds a match, or null if not. So for this function we could use 
```
  return string.match(/[aeiou]/i);
```
and it would return `true` if the string contained a lower or upper case vowel, and `null` if not. But since we only want a `true` or `false` value to be returned, we can use that double negation (!!) method again to convert the `null` value to false:
```
  return !!string.match(/[aeiou]/i);
```
Way more concise.